File: tests/fixtures/kotlin-strategy-pos/app/src/test/java/app/PricingTest.kt
package app

import org.junit.Test
import org.junit.Assert.assertEquals

class PricingTest {
    @Test fun `regular price unchanged`() = assertEquals(100.0, price("regular", 100.0), 0.001)
    @Test fun `vip gets 20 percent off`() = assertEquals(80.0, price("vip", 100.0), 0.001)
    @Test fun `staff gets 50 percent off`() = assertEquals(50.0, price("staff", 100.0), 0.001)
    @Test fun `discounted total for vip`() = assertEquals(160.0, discountedTotal("vip", 100.0, 2), 0.001)
}

File: tests/fixtures/dart-strategy-neg/test/pricing_test.dart
import 'package:test/test.dart';
import 'package:dart_strategy_neg/pricing.dart';

void main() {
  test('regular price unchanged', () => expect(price('regular', 100), 100.0));
  test('vip gets 20% off', () => expect(price('vip', 100), 80.0));
  test('staff gets 50% off', () => expect(price('staff', 100), 50.0));
  test('discounted total for vip', () => expect(discountedTotal('vip', 100, 2), 160.0));
}
